src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -34,100 +38,165 @@ object StructuralVariant {
   }
 
   case class Location(
-      contig: String,
-      position: Long) {
-  }
+    contig: String,
+    position: Long)
 
   case class GenomeRange(
     contig: String,
     start: Long,
     stop: Long)
 
+  case class MedianStats(
+    median: Double,
+    mad: Double)
+
+  // An undirected, weighted graph of evidence for structural variants.
+  // Nodes are paired reads.
+  // Edges represent compatibility between the induced structural variants,
+  // with the weight being the strength of compatibility.
+  type PairedReadGraph = Graph[PairedMappedRead, WUnDiEdge]
+
   object Caller extends SparkCommand[Arguments] {
     override val name = "structural-variant"
     override val description = "Find structural variants, e.g. large deletions."
 
     // The sign of inferredInsertSize follows the orientation of the read. This returns
     // an insert size which is positive in the common case, regardless of the orientation.
-    def orientedInsertSize(r: PairedRead[MappedRead]): Option[Int] = {
-      val sgn = if (r.isPositiveStrand) { 1 } else { -1 }
-      r.mateAlignmentProperties.flatMap(_.inferredInsertSize).map(x => x * sgn)
+    def orientedInsertSize(r: PairedMappedRead): Int = {
+      val sgn = if (r.read.isPositiveStrand) { 1 } else { -1 }
+      r.inferredInsertSize * sgn
     }
 
-    // Given a sorted sequence of positions (all of which are multiples of @blockSize), coalesce ranges of
-    // consecutive multiples of @blockSize into tuples representing the start and end (inclusive) of each range.
-    // For example, (0, 10, 20, 30, 70, 80) --> [(0, 30), (70, 80)]
-    def coalesceAdjacent(xs: IndexedSeq[Long], blockSize: Long): Iterator[(Long, Long)] = {
-      val rangeStarts = ArrayBuffer(0)
-      for (i <- 1 until xs.length) {
-        if (xs(i - 1) + blockSize != xs(i)) {
-          rangeStarts += i
-        }
+    // Extract median and MAD (Median Absolute Deviation) from an unordered Array.
+    def medianStats[T <% Double](xs: Array[T])(implicit ord: Ordering[T]): MedianStats = {
+      if (xs.isEmpty) {
+        return MedianStats(0.0, 0.0)
```

Comment:
Not sure of others' thoughts on this but I think `return`s like this are somewhat un-idiomatic in Scala; I remember hearing in particular that `return` is implemented by `throw`ing, which is perf-costly.

```
if (xs.isEmpty) {
  MedianStats(0.0, 0.0)
} else {
  …
}
```

or

```
xs match {
  case Array(head, rest @ _*) => 
    …
  case _ => 
    MedianStats(0.0, 0.0)
}
```

maybe better?